homeassistant/components/snapcast/media_player.py

```diff
@@ -275,6 +279,68 @@ async def async_unjoin(self) -> None:
         """Handle the unjoin service."""
         raise NotImplementedError
 
+    def _get_metadata(self, key, default=None) -> Any:
```

Comment:
Can we type this better?

Reply:
Gave it a shot. Not sure if there's a better option than `Mapping[str, str | list[str | None]]`